servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -1497,16 +1495,31 @@ public final Completable ambWith(final Completable other) {
     // Conversion Operators End
     //
 
+    /**
+     * Returns the {@link AsyncContextMap} to be used for a subscribe.
+     *
+     * @param provider The {@link AsyncContextProvider} which is the source of the map
+     * @return {@link AsyncContextMap} for this subscribe operation.
+     */
+    AsyncContextMap contextForSubscribe(AsyncContextProvider provider) {
+        // the default behavior is to copy the map. Some operators may want to use shared map
+        return provider.contextMap().copy();
+    }
+
     /**
      * A internal subscribe method similar to {@link CompletableSource#subscribe(Subscriber)} which can be used by
      * different implementations to subscribe.
      *
      * @param subscriber {@link Subscriber} to subscribe for the result.
      */
     protected final void subscribeInternal(Subscriber subscriber) {
-        AsyncContextProvider provider = AsyncContext.provider();
-        subscribeWithContext(subscriber, provider,
-                shareContextOnSubscribe() ? provider.contextMap() : provider.contextMap().copy());
+        AsyncContextProvider contextProvider = AsyncContext.provider();
+        AsyncContextMap contextMap = contextForSubscribe(contextProvider);
+        if (LOGGER.isTraceEnabled()) {
```

Comment:
this seems like it is primarily for debugging ... does it make sense to remove for now?

Reply:
Sure.